published/004-2016-Jan-25.md

```diff
@@ -0,0 +1,92 @@
+# IPFS Weekly #4
+
+[IPFS](//ipfs.io/) is a new hypermedia distribution protocol, addressed by content and identities, aiming to make the web faster, safer, and more open. In these posts, we highlight some of the development that has happened in the past week. For anyone looking to get involved, follow the embedded hyperlinks, search the wealth of information on [GitHub](//github.com/ipfs) or join us on [IRC](//webchat.freenode.net/?channels=ipfs) (#ipfs on the Freenode network).
+
+If you would like to get this update as an email, sign up for our [weekly newsletter](//tinyletter.com/ipfsweekly)!
+
+Here are some of the highlights for the [January 25th](//github.com/ipfs/pm/issues/84) sprint:
+
+## Updates
+
+* [**js-ipfs**](//github.com/ipfs/js-ipfs): [@diasdavid](//github.com/diasdavid) created js-ipfs-merkle-dag and js-ipfs-blocks, which will help IPFS deal with MerkleDAGs and are extensible enough to allow others to build their own MerkleDAG structures. Also, js-ipfs is now able to import a file into the MerkleDAG, an important milestone. An architecture graph has been spec'ed out [here](https://github.com/ipfs/js-ipfs#ipfs-core-implementation-architecture) to help newcomers understand how js-ipfs works. Finally, lots of new issues and low-hanging fruit have been tagged for [contributors to tackle](//github.com/ipfs/js-ipfs/issues).
+* [**idb-plus-blob-store**](https://github.com/dignifiedquire/idb-plus-blob-store): Node streams are not able to provide consistent guarantees for writes on the "end" event, so [@dignifiedquire](//github.com/dignifiedquire) found a way to make them do that so that we can use them in js-ipfs. This fixes some inconsistencies that substack/idb-blob-store has.
```

Comment:
[**idb-plus-blob-store**](https://github.com/dignifiedquire/idb-plus-blob-store): Node streams are not able to provide consistent guarantees for writes on the "end" event, so [@dignifiedquire](//github.com/dignifiedquire) found a way to make them do that so that we can use them in js-ipfs. This fixes some inconsistencies that substack/idb-blob-store has.

[**idb-plus-blob-store**](https://github.com/dignifiedquire/idb-plus-blob-store): The [substack/idb-blob-store](https://github.com/substack/idb-blob-store) had issues regarding emitting the `finish` event on the `createWritableStream` result due to internal issues in streams and the way `.end` is handled. This module fixes that and now allows us to test all parts of js-ipfs in the browser using IndexedDB as the storage.